src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java

```diff
@@ -307,9 +307,28 @@ public void endReport(List<Generator> adjustableGenerators) {
                         .forEach(g -> report(reporter, suffixKey, "GeneratorUnchangedTargetP", "Generator ${generator} has not been selected by the merit order algorithm. Its active power set point has been set to 0",
                                 Map.of(GENERATOR, g.getId()), TypedValue.TRACE_SEVERITY));
             }
+            // report the max marginal cost used
+            Double maxUsedMarginalCost = getGeneratorsMarginalCost(updatedGenerators).stream()
+                    .mapToDouble(Double::doubleValue)
+                    .max()
+                    .getAsDouble();
+
+            report(reporter, suffixKey, "MaxUsedMarginalCost", "Marginal cost: ${maxUsedMarginalCost}",
+                    Map.of("maxUsedMarginalCost", maxUsedMarginalCost), TypedValue.INFO_SEVERITY);
         }
     }
 
+    private static List<Double> getGeneratorsMarginalCost(List<Generator> generators) {
+        List<Double> marginalCosts = new ArrayList<>();
+        generators
+                .forEach(generator -> {
+                    if (getGeneratorMarginalCost(generator) != null) {
+                        marginalCosts.add(getGeneratorMarginalCost(generator));
```

Comment:
There are still 2 calls to getGeneratorMarginalCost

Reply:
Fixed. No need for this method anymore.